Bottom: 11041dc830cb6654bab95b4d725bba8e8a055535
Top:    ed95af8a8dadd49110629bba504ae22d5ccb2307
Author: Jason Wang <jasowang@redhat.com>
Date:   2015-01-27 03:32:37 -0500

virtio-net: should use netif_wakeup_queue()

Which can reschedule qdisc immediately.

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 8f60cf2..91fed38 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -234,8 +234,8 @@ static unsigned int free_old_xmit_skbs(struct netdev_queue *txq,
 		packets++;
 	}
 
-	if (sq->vq->num_free >= 2+MAX_SKB_FRAGS)
-		netif_tx_start_queue(txq);
+	if (sq->vq->num_free >= 2 + MAX_SKB_FRAGS)
+		netif_wake_subqueue(vi->dev, vq2txq(sq->vq));
 
 	return packets;
 }
